firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -131,10 +131,16 @@ def __init__(self, *parts):
                 raise ValueError(error)
         self.parts = tuple(parts)
 
-    @staticmethod
-    def from_string(string):
+    @classmethod
+    def from_string(cls, string):
         """ Creates a FieldPath from a unicode string representation.
 
+        The constructor must be used if the unicode string representation
+        consists of invalid characters `~*/[]`. This method always splits
+        on the character `.` for nesting. If any field names in this field
+        path includes the character `.`, the constructor must also be used
```

Comment:
You still have the phrase "the constructor must also be used to determine nesting", which I don't understand. I don't think "nesting" is a good word to use here.

Let's try:

"This method splits on the character `.` and disallows the characters `~*/[]`. To create a FieldPath whose components have those characters, call the constructor."

Reply:
That sounds to me like you can't use this function to call FieldPath.from_string('a.b'), which creates the api_repr `a.b`.  But you won't be able to use it to do `(bt)a.b(bt)`

Reply:
It's a FieldPath whose *components* have those characters. The components of `FieldPath.from_string('a.b')` are `a` and `b`, which have no invalid characters or dots.